log4j-appender/src/test/java/org/apache/kafka/log4jappender/MockKafkaLog4jAppender.java

```diff
@@ -42,7 +46,18 @@ protected void append(LoggingEvent event) {
         super.append(event);
     }
 
-    protected List<ProducerRecord<byte[], byte[]>> getHistory() {
+    @Override
+    void getResponse(Future<RecordMetadata> response) throws InterruptedException, ExecutionException {
+        if (failSync)
+            throw new InterruptedException("Failed");
```

Comment:
You could use MockProducer.errorNext()

Reply:
It can be done only in a separate thread, just before logging call